tests/test_predictors/test_svm_word_predictor.py

```diff
@@ -0,0 +1,286 @@
+
+
+class TestSVMWordPredictor(unittest.TestCase):
+    @classmethod
+    def setUp(self):
+        self.fixture_path = os.path.join(os.path.dirname(__file__), "../fixtures/")
+        self.predictor = SVMWordPredictor.from_path(
+            tar_path=os.path.join(
+                self.fixture_path, "svm_word_predictor/svm_word_predictor.tar.gz"
+            )
+        )
+        with open(
+            os.path.join(
+                self.fixture_path,
+                "types/20fdafb68d0e69d193527a9a1cbe64e7e69a3798__pdfplumber_doc.json",
+            )
+        ) as f_in:
+            doc_dict = json.load(f_in)
+            self.doc = Document.from_json(doc_dict=doc_dict)
+
+    def test_predict(self):
+        words = self.predictor.predict(document=self.doc)
+        # double-check number of units
+        self.assertLess(len(words), len(self.doc.tokens))
+        # double-check can annotate
+        self.doc.annotate(words=words)
+        # after annotating, double-check words against tokens
+        for word in self.doc.words:
+            tokens_in_word = word.tokens
+            # if word is a single token, then it should be the same as the token
+            if len(tokens_in_word) == 1:
+                self.assertEqual(tokens_in_word[0].text, word.text)
+                self.assertEqual(tokens_in_word[0].spans, word.spans)
+            else:
+                # otherwise, most token units should be substring of word
+                overlap = np.mean([token.text in word.text for token in tokens_in_word])
+                self.assertGreaterEqual(overlap, 0.5)
+                # spans should match up though
+                self.assertEqual(
+                    [span for token in tokens_in_word for span in token.spans],
+                    word.spans,
+                )
+        # uncomment for debugging:
+        for word in self.doc.words:
+            token_text = str([t.text for t in word.tokens])
+            if len(word.tokens) > 1 and "-" in token_text:
+                print(f"{token_text}\t-->\t{word.text}")
+
+    def test_validate_tokenization(self):
+        doc = Document.from_json(
+            doc_dict={
+                "symbols": "I am the wizard-of-oz.",
+                "tokens": [
+                    {"id": 0, "spans": [{"start": 0, "end": 1}]},
+                    {"id": 1, "spans": [{"start": 2, "end": 4}]},
+                    {"id": 2, "spans": [{"start": 5, "end": 8}]},
+                    {"id": 3, "spans": [{"start": 9, "end": 15}]},
+                    {"id": 4, "spans": [{"start": 15, "end": 16}]},
+                    {"id": 5, "spans": [{"start": 16, "end": 18}]},
+                    {"id": 6, "spans": [{"start": 18, "end": 19}]},
+                    {"id": 7, "spans": [{"start": 19, "end": 21}]},
+                    {"id": 8, "spans": [{"start": 21, "end": 22}]},
+                ],
+            }
+        )
+        self.predictor._validate_tokenization(document=doc)
+        # missing token id
```

Comment:
what's the scenario where this could happen?

Reply:
someone reconstructs a document from scratch without an `id` field

one thought is -- this is starting to become a requirement for everything; maybe it should be baked into the `Document.from_json()` method that if `id` isnt specified, it gives a `warning` and automatically assigns an `id`?

what do u think

Reply:
> I'm also wondering: to my eye, this still relies on an association of token id and word ids. I know the problem with the original predictor was this violated assertion: `assert None not in token_id_to_word_id.values()` Did we change something so that this error can't happen?

great catch. we changed 2 things:
* different implementation. the previous approach required this iterative "building up" of the `token_id_to_word_id` map, and this assertion was necessary to make sure we didn't miss writing rules for certain cases when building this up.  the current implementation takes a different approach to this incremental build-up. it automatically assumes all tokens are words themselves, so it seeds `token_id_to_word_id` with non-null values from the get-go. 
* different ideology. i realized visiting this problem properly that we should absolutely allow for tokens to be skipped, if necessary, which would mean there would be tokens mapped to no word. this implementation doesn't do this, but conceptually, we need to allow for tokens mapped to nothing if we want to future handle cases like hallucinated/fake tokens in PDFs